synapse/storage/databases/main/metrics.py

```diff
@@ -521,3 +521,30 @@ def _generate_user_daily_visits(txn):
         await self.db_pool.runInteraction(
             "generate_user_daily_visits", _generate_user_daily_visits
         )
+
+
+class UserMetricsStore(SQLBaseStore):
+    def __init__(self, database: DatabasePool, db_conn, hs: "HomeServer"):
+        super().__init__(database, db_conn, hs)
+
+    async def count_total_messages(self, user_id):
+        """
+        Counts the total number of messages sent by a user.
+        """
+
+        def _count_messages(txn):
+
+            sql = """
+                SELECT COALESCE(COUNT(*), 0) FROM events
+                WHERE type = 'm.room.message'
+                    OR type = 'm.room.encrypted'
+                    AND sender LIKE ?
+            """
```

Comment:
I think it would be wiser to make the list of types to retrieve an argument of the function, that way modules would have more flexibility on what event is considered in the count (and we could potentially save up on performance by counting less events).

Also, we probably want `sender = ?` instead of `LIKE`, I can't see any reason why `user_id` would be a SQL regex.